src/WP_CLI/CommandWithTranslation.php

```diff
@@ -34,64 +34,78 @@ public function update( $args, $assoc_args ) {
 			return;
 		}
 
-		// Gets a list of all languages.
-		$all_languages = $this->get_all_languages();
-
-		$updates_per_type = array();
-
-		// Formats the updates list.
-		foreach ( $updates as $update ) {
-			if ( 'plugin' === $update->type ) {
-				$plugins	 = get_plugins( '/' . $update->slug );
-				$plugin_data = array_shift( $plugins );
-				$name		 = $plugin_data['Name'];
-			} elseif ( 'theme' === $update->type ) {
-				$theme_data	 = wp_get_theme( $update->slug );
-				$name		 = $theme_data['Name'];
-			} else { // Core
-				$name = 'WordPress';
-			}
-
-			// Gets the translation data.
-			$translation = wp_list_filter( $all_languages, array(
-				'language' => $update->language
-			) );
-			$translation = (object) reset( $translation );
+		// Only preview which translations would be updated.
+		if ( Utils\get_flag_value( $assoc_args, 'dry-run' ) ) {
+			WP_CLI::line( sprintf( 'Found %d translation updates that would be processed:', count( $updates ) ) );
```

Comment:
Shouldn't this be a count per type?

This would mean the `--dry-run` check and early return needs to happen _after_ the foreach loop that builds the `$updates_per_type` array.

Reply:
Hmm yeah I somehow moved this from _after_ the loop to _before_ 🤦‍♀️ ☕️